instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/TemporaryMetricsView.java

```diff
@@ -52,10 +44,72 @@
     return view;
   }
 
-  static Attributes applyDurationView(Attributes startAttributes, Attributes endAttributes) {
+  static Attributes applyClientDurationView(Attributes startAttributes, Attributes endAttributes) {
+    Set<AttributeKey> fullSet = new HashSet<>(durationAlwaysInclude);
+    // We pull identifying attributes according to:
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attribute-alternatives
+    if (containsAttribute(SemanticAttributes.HTTP_URL, startAttributes, endAttributes)) {
+      // Use http.url alone
+      fullSet.add(SemanticAttributes.HTTP_URL);
+    } else if (containsAttribute(SemanticAttributes.HTTP_HOST, startAttributes, endAttributes)) {
+      // Use http.scheme, http.host and http.target
+      fullSet.add(SemanticAttributes.HTTP_SCHEME);
+      fullSet.add(SemanticAttributes.HTTP_HOST);
+      fullSet.add(SemanticAttributes.HTTP_TARGET);
```

Comment:
None of our HTTP client instrumentations set these 3 attributes - they always set only `http.url` (see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractor.java#L61)

Reply:
Same question for server.   Guarantee or current implementation?

Reply:
Guaranteed for all instrumentation that we write in this repository. And the Instrumenter API is (recently) very opinionated about this.

Reply:
Updated based on our SiG discussion